### DISCUSSION TITLE
Errors changing folder location of servers data files

### DISCUSSION THREAD

**Discussion by LeoKryssen** [pterodactyl/panel#5219] [Support] [answered]
Hi, 
I'm pretty new to docker and pterodactyl, but i need your help
I'm running both pterocactyl and wings inside docker, with compose, files both available below.

I trides to run a paper minecraft server at first, without changing any of the directories. Files go to /var/lib/pterocatyl/volumes without issues. Everything went well, the server starts, I can connect to it.

But for storage issues, I need to put files to another drive, so I changed the "Daemon Server File Directory" to my other drive, as : /mnt/storage/pterodactyl/volumes. Changed the bind mount of the docker compose of wings to that location too. But then, I encounter two errors.

The first tells me the "bind" is invalid :

> container@pterodactyl~ Error Event [5e7d9a7e-ba53-4993-b14b-71b2c5bdd9ef]: environment/docker: failed to run pre-boot process: environment/docker: failed to create container: Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /mnt/storage/pterodactyl/volumes/357b9380-6b06-4ca2-b63b-21a135363024

Same error when selecting a folder from the boot disk :

> container@pterodactyl~ Error Event [5e7d9a7e-ba53-4993-b14b-71b2c5bdd9ef]: environment/docker: failed to run pre-boot process: environment/docker: failed to create container: Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /home/leokryssen/pterodactyl/volumes/357b9380-6b06-4ca2-b63b-21a135363024

The second, when i manually create the folders, tells me the daemon can't access to the .jar file

> container@pterodactyl~ java -Xms128M -XX:MaxRAMPercentage=95.0 -Dterminal.jline=false -Dterminal.ansi=true -jar server.jar
Error: Unable to access jarfile server.jar

No files are created in the final folder, not like in /var/lib/pterodactyl/volumes
I checked the permissions, tried to chmod 700 and chown to root:root or 988:988 but still stucked at that last error

Here is the pterodactyl compose :
```dockerfile
version: "3.8"
x-common:
  database:
    # Do not remove the "&db-password" from the end of the line below, it is important
    # for Panel functionality.
    &a1
    MYSQL_PASSWORD: ToshiieShinZhao430
    MYSQL_ROOT_PASSWORD: ZhaoYun430
  panel:
    # This URL should be the URL that your reverse proxy routes to the panel server
    &a2
    APP_URL: http://pterodactyl.leokryssen.fr
    # A list of valid timezones can be found here: http://php.net/manual/en/timezones.php
    APP_TIMEZONE: UTC
    APP_SERVICE_AUTHOR: admin@leokryssen.fr
    TRUSTED_PROXIES: "*" # Set this to your proxy IP
    # Uncomment the line below and set to a non-empty value if you want to use Let's Encrypt
    # to generate an SSL certificate for the Panel.
    # LE_EMAIL: ""
  mail: &a3
    MAIL_FROM: admin@leokryssen.fr
    MAIL_DRIVER: smtp
    MAIL_HOST: mail.leokryssen.fr
    MAIL_PORT: "465"
    MAIL_USERNAME: admin@leokryssen.fr
    MAIL_PASSWORD: ToshiieShinZhao430
    MAIL_ENCRYPTION: "true"
#
# ------------------------------------------------------------------------------------------
# DANGER ZONE BELOW
#
# The remainder of this file likely does not need to be changed. Please only make modifications
# below if you understand what you are doing.
#
services:
  database:
    image: mariadb:10.5
    restart: always
    command: --default-authentication-plugin=mysql_native_password
    volumes:
      - /srv/pterodactyl/database:/var/lib/mysql
    environment:
      <<: *a1
      MYSQL_DATABASE: panel
      MYSQL_USER: pterodactyl
  cache:
    image: redis:alpine
    restart: always
  panel:
    image: ghcr.io/pterodactyl/panel:latest
    restart: always
    ports:
      - 81:80
      # - 443:443
    links:
      - database
      - cache
    volumes:
      - /srv/pterodactyl/var/:/app/var/
      - /srv/pterodactyl/php/:/var/www/pterodactyl/
      - /srv/pterodactyl/nginx/:/etc/nginx/http.d/
      - /srv/pterodactyl/certs/:/etc/letsencrypt/
      - /srv/pterodactyl/logs/:/app/storage/logs
      - /home/leokryssen/pterodactyl/:/var/lib/pterodactyl/ #remove if new file created
    environment:
      <<:
        - *a2
        - *a3
      DB_PASSWORD: ToshiieShinZhao430
      APP_ENV: production
      APP_ENVIRONMENT_ONLY: "false"
      CACHE_DRIVER: redis
      SESSION_DRIVER: redis
      QUEUE_DRIVER: redis
      REDIS_HOST: cache
      DB_HOST: database
      DB_PORT: "3306"
networks:
  default:
    ipam:
      config:
        - subnet: 172.20.0.0/16

``` 
And the wings compose : 

```dockerfile
version: "3.8"
services:
  wings:
    image: ghcr.io/pterodactyl/wings:latest
    restart: always
    networks:
      - wings0
    ports:
      - 8080:8080
      - 2022:2022
    tty: true
    environment:
      TZ: UTC
      WINGS_UID: 988
      WINGS_GID: 988
      WINGS_USERNAME: pterodactyl
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /var/lib/docker/containers/:/var/lib/docker/containers/
      - /etc/pterodactyl/:/etc/pterodactyl/
      - /home/leokryssen/pterodactyl/volumes/:/var/lib/pterodactyl/volumes/
      - /var/log/pterodactyl/:/var/log/pterodactyl/
      - /tmp/pterodactyl/:/tmp/pterodactyl/
      - /etc/ssl/certs:/etc/ssl/certs:ro
networks:
  wings0:
    name: wings0
    driver: bridge
    ipam:
      config:
        - subnet: 172.21.0.0/16
    driver_opts:
      com.docker.network.bridge.name: wings0

``` 

Maybe I'm doing something wrong, but i don't know what

**Comment by danny6167** [comment, marked as answer]
Your issue is here. ```/home/leokryssen/pterodactyl/volumes/:/var/lib/pterodactyl/volumes/```
You can not remap the servers data folder like that. The path must be the same inside and outside the wings container.

If you want to store servers in `/home/leokryssen/pterodactyl/volumes` then you must map that directory to `/home/leokryssen/pterodactyl/volumes` inside the wings container and use `/home/leokryssen/pterodactyl/volumes` as the path in your wings `config.yml`

**Reply by Buco7854** [reply]
Hello,

But what if I use a docker named volume?

I have `data:/var/lib/pterodactyl`

I personally store everything in docker volumes is there any work around? Can I even do it by putting the path to where docker volumes are supposed to be stored `/var/lib/docker/volumes/pterodactyl-wings-wings-1/` (shoudl i add `_data`) or am i blocked?

Thanks
